core/blockchain.go

```diff
@@ -347,6 +347,12 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis
 	// Make sure the state associated with the block is available, or log out
 	// if there is no available state, waiting for state sync.
 	head := bc.CurrentBlock()
+	// Make sure Cel2 chain has migrated data from Celo1
+	if chainConfig.Cel2Time != nil && !chainConfig.IsCel2(head.Time) {
```

Comment:
Is this check sufficient or should we also check for the expected transformations in the migrated data? For example, we could check that the Header has an Uncles field. This would make the check dependent on not just the Cel2Time in the config but also on the actual data. Maybe this is overkill though, just a thought.

Reply:
@alecps 

This quick check makes sure the node has migrated data for Cel2 blockchain and tells a node operator missing migrated data quickly. With this code, it assumes the migrated data is valid.

I think it's the operator's responsibility to supply accurate migrated data. Such data validation will be expensive in bootstrapping. I would do it offline if necessary.